telemetry/definitions/commonDefinitions.json

```diff
@@ -850,6 +905,86 @@
             "name": "apprunner_viewServiceLogs",
             "description": "View the App Runner service logs (the logs produced by App Runner)"
         },
+        {
+            "name": "auth_gettingStartedPage",
+            "description": "Called when user lands on getting started page",
+            "metadata": [
+                { "type" : "source" },
+                { "type" : "numberOfConnections"}
+            ] 
+        },
+        {
+            "name": "auth_connectToAWS",
+            "description": "Called when user clicks on getting ConnectToAWS from any source. e.g. explorer, developer tools, etc",
+            "metadata": [
+                { "type": "source" },
+                { "type": "numberOfConnections"}
+            ]
+        },
+        {
+            "name": "auth_awsResources",
+            "description": "Called upon user interacting with View, Modify, and Deploy AWS Resources Tab",
+            "metadata": [
+                { "type": "profileNameStatus"},
+                { "type": "accessKeyStatus"},
+                { "type": "secretStatus"},
+                { "type": "iamIdentityCenterStartURL"}, 
+                { "type": "iamIdentityCenterRegion", "required": false },
+                { "type": "errorMsg", "required": false},
+                { "type": "result"}
+            ]
+        },
+        {
+            "name": "auth_codeWhisperer",
+            "description": "Called when user clicks AI-powered Code suggestions from Codewhisperer tab under getting started. Since there are two ways of loggining we want to know status of each available option",
+            "metadata": [
+                { "type": "builderIDStatus"},
+                { "type": "iamIdentityCenterStartURL"}, 
+                { "type": "iamIdentityCenterRegion", "required": false },
+                { "type": "errorMsg", "required": false},
+                { "type": "result"}
+            ]
+        },
+        {
+            "name": "auth_codeCatalyst",
+            "description": "Called when user clicks Launch CodeCatalyst Cloud-based Environments tab under getting started",
+            "metadata": [
+                { "type": "builderIDStatus"},
+                { "type": "result"}
+            ]
+        },
+        {
+            "name": "auth_signUpForFree", 
+            "description": "Called upon user clicking sign up for free in IAM Credentials",
+            "metadata": [
+                { "type": "result"}, // can we know if the user came back signing up successfully?
+                { "type": "source"}
+            ]
+        },
+        {
+            "name": "auth_addIAMUserCredentials",  // redundant
+            "description": "Called when user clicks add IAM User Credentials",
+            "metadata": [
+                { "type": "source"},  // do we wanna do status check for each input field here? 
+                { "type": "result"}
+            ]
```

Comment:
If we want to keep track success rate of each authentication method. is it good idea to emit separate metrics? Or we skip this granularity because are covering bigger picture in auth_codeCatalyst, auth_codeWhisperer and auth_awsResources tab.

Reply:
I'd use separate definitions if they are substantially different or if the semantics of a field are incompatible. I wouldn't worry about needing to "scale" a definition too much. Telemetry data (and its interpretation!) is closely tied to the actual implementation. If a definition needs lots of changes, it's probably wise to create a new definition instead.

Reply:
Yeah. I think in this case it makes more sense to have auth success different as other metrics don't really define this. and we should avoid interpretations.